EIPS/eip-4910.md

```diff
@@ -0,0 +1,973 @@
+
+<a name="r58">**[R58]**</a> *The public `royaltyPayOut` function interface MUST adhere to the definition below:*
+```
+/// @dev Function to payout a royalty payment
+/// @param tokenId is the identifier of the NFT token
+/// @param RAsubaccount is the address of the Royalty Sub Account from which the payout should happen
+/// @param receiver is the address to receive the payout
+/// @param amount is the amount to be paid out
+
+function royaltyPayOut (uint256 tokenId, address RAsubaccount, address payable payoutAccount, uint256 amount) public virtual nonReentrant returns (bool)
+```
+The Boolean return value is `true` for a successful function execution, and `false` for an unsuccessful function execution.
+
+Note, that the function has `reentrancy` protection through `nonReentrant` from the Open Zeppelin library since funds are being paid out.
+
+<a name="r59">**[R59]**</a> *The input parameters of the `royaltyPayOut` function MUST satisfy the following requirements:*
+* *`_msgSender() == RAsubaccount`.*
+* *`tokenId` must exist and MUST NOT be burned.*
+* *`tokenId` must be associated with a Royalty Account.*
+* *`RAsubaccount` must be a valid `accountId` in a Royalty Sub Account of the Royalty Account of the `tokenId'.*
+* *`isIndividual == true` for the Royalty Sub Account, `RAsubaccount`.*
+* *`amount <= royaltybalance` of the Royalty Sub Account, `RAsubaccount.*`
+
+<a name="r60">**[R60]**</a> *The internal `_royaltyPayOut` function interface MUST adhere to the definition below*:
+```
+function _royaltyPayOut (uint256 tokenId, address RAsubaccount, address payable payoutAccount, uint256 amount) public virtual returns (bool)
+```
+
+<a name="r61">**[R61]**</a> *The internal `_royaltyPayOut` function MUST perform the following actions:
+* *send the payment to the `payoutaccount`.*
+* *update the `royaltybalance` of the `RAsubaccount` of the Royalty Account upon successful transfer.*
+
+<a name="r62">**[R62]**</a> *The following steps MUST be taken to send out a royalty payment to its recipient:*
+* *find the Royalty Sub Account.*
+* *extract `tokentype` from the Royalty Sub Account.*
+* *based on the token type send to the `payoutAccount` either*
+    * *'ETH' / relevant protocol token or*
+    * *another token based on token type* 
+* *and only if the payout transaction is successful, deduct `amount` from `royaltybalance` of the Royalty Sub Account,`RAsubaccount`, and then return `true` as the function return parameter, otherwise return `false`.* 
+
+## Backwards Compatibility
+This EIP is backwards compatible to the ERC721 standard introducing new interfaces and functionality but retaining the core interfaces and functionality of the ERC-721 standard.
+
+## Test Cases
+A full test suite is part of the reference implementation.
+
+## Reference Implementation
+The Treetrunk reference implementation of the standard can be found [here](https://github.com/ethereum/EIPs/tree/master/assets/eip-4910/eip-4910-reference-implementation/).
```

Comment:
This link won't work as directory listing is not supported.  You need to link to each individual file from here.

Reply:
@MicahZoltu ok ... will not link directly.

Reply:
Could you elaborate on why a directory listing isn't supported?

Reply:
removed reference implementation